### DISCUSSION TITLE
Aggregate changed: How to handle snapshot store

### DISCUSSION THREAD

**Discussion by trancefreak77** [rasmus/EventFlow#931] [Q&A]
EventFlow has the concept of event upgraders when a new version of an event looks different. Is there also a concept with regards to the snapshot store?

If a new version of an event needs to be implemented the aggregate may also change. I am using the EF Core snapshot store implementation. Do I have to delete the snapshot store in this case? How can I repopulate the snapshot store?

Is there an existing concept / solution for this kind of issue?

Regards,
Christian

**Comment by rasmus** [comment]
Hi @trancefreak77 you can implement a ` ISnapshotUpgrader` to upgrade your snapshots

https://github.com/eventflow/EventFlow/blob/develop-v1/Source/EventFlow/Snapshots/ISnapshotUpgrader.cs

**Comment by trancefreak77** [comment]
Hi Rasmus, thanks very much, thats exactly what I was looking for.
